Support for scram-sha-256 authentication method

zwirek opened the pull request.

zwirek committed:
Support for SCRAM_SHA_256

mbonneau commented on tests/Integration/ScramSha256PasswordTest.php:
```diff
@@ -0,0 +1,43 @@
+
+class ScramSha256PasswordTest extends TestCase
+{
+    public function testScamSha256Login()
```
This method name is missing an 'r'

zwirek replied:
This is definitely not a scam :)

mbonneau reviewed (commented):
This looks good. Thanks for taking the time to work on this. I did not have a chance to look at why the test was timing out.

zwirek committed:
update postgres config for CI tests to handle scram-sha-256 auth along with md5

zwirek committed:
remove not supported in php7.0 void return type from ScramSha256

zwirek committed:
Fix ClientTest mock creation to be compatible with phpunit 10

zwirek commented on src/PgAsync/Message/Message.php:
```diff
@@ -23,11 +23,11 @@ public static function prependLengthInt32(string $s): string
     }
 
-    public static function createMessageFromIdentifier(string $identifier): ParserInterface
+    public static function createMessageFromIdentifier(string $identifier, array $dependencies): ParserInterface
```
I have taken this as an internal method but you are right. This should have default value

zwirek commented:
Thanks for the review. My overlook CI configuration is causing the failed tests. The only thing I have changed is docker-compose to have an extra service with postgres 15 for local development

In order to tests start to pass, I have changed CI configuration and setup postgres to have additional user with scram-sha-256 authorization method. I have added additional postgres versions as 15 is the latest.

Neustradamus commented:
@zwirek: Really a good job! Thanks.

And @mbonneau too for reviewing!
